tests/_multi_screen.lua

```diff
@@ -0,0 +1,509 @@
+-- Helper module to replicate a set of steps across multiple screen scenarios
+-- Ideas for improvments:
+--  * Export an iterator so the suits can call the screen change step by step
+--    themselves instead of the current all or nothing
+--  * Offer multiple scenarios such as screen add, remove, resize/swap instead
+--    of having to test all corner cases all the time.
+local wibox = require("wibox")
+local surface = require("gears.surface")
+local wallpaper = require("gears.wallpaper")
+local color = require("gears.color")
+local shape = require("gears.shape")
+
+local module = {}
+
+-- Get the original canvas size before it is "lost"
+local canvas_w, canvas_h = screen[1].geometry.width, screen[1].geometry.height
+
+-- Create virtual screens.
+-- The steps will be executed on each set of screens.
+local dispositions = {
+    -- A simple classic, 2 identical screens side by side
+    {
+        function()
+            return {
+                x = 0,
+                y = 0,
+                width  = math.floor(canvas_w/2),
+                height = canvas_h,
+            }
+        end,
+        function()
+            return {
+                x = math.floor(canvas_w/2)+1,
+                y = 0,
+                width = math.floor(canvas_w/2),
+                height = canvas_h,
+            }
+        end,
+    },
+
+    -- Take the previous disposition and swap the screens
+    {
+        function()
+            return {
+                x = math.floor(canvas_w/2)+1,
+                y = 0,
+                width = math.floor(canvas_w/2),
+                height = canvas_h,
+            }
+        end,
+        function()
+            return {
+                x = 0,
+                y = 0,
+                width  = math.floor(canvas_w/2),
+                height = canvas_h,
+            }
+        end,
+        keep = true,
+    },
+
+    -- Take the previous disposition and resize the leftmost one
+    {
+        function()
+            return {
+                x = math.floor(canvas_w/4)+1,
+                y = 0,
+                width = math.floor(3*(canvas_w/4)),
```

Comment:
How about having some "global variables" `local canvas_w_half = math.floor(canvas_w/2)` etc (`canvas_w_quarter`, `canvas_w_three_quarter`, ...). I'm not totally sure, but it might improve readability by getting of all the rounding that is needed here.